text_scrubber/geo/geo.py

```diff
@@ -0,0 +1,336 @@
+    country_match = find_closest_string(cleaned_country, _COUNTRY_RESOURCES['cleaned_trigrams'])
+    if country_match:
+        best_matches, score = country_match
+        candidates = (_COUNTRY_RESOURCES['cleaned_to_capitilized'][country] for country in best_matches)
+        return [(candidate, score) for candidate in candidates] if return_scores else list(candidates)
+
+    # No match found
+    return []
+
+
+def normalize_state(state: str, return_scores: bool = False) -> List[Union[Tuple[str, str, float], Tuple[str, str]]]:
+    """
+    Cleans up a state by string cleaning and performs some basic state lookups to get the canonical name.
+
+    :param state: State name or code.
+    :param return_scores: Whether to return match scores.
+    :return: List of (state, country) candidates in canonical form.
+    """
+    # Clean state
+    cleaned_state = clean_state(state)
+
+    # Check if state is part of the known states list
+    if cleaned_state in _STATE_RESOURCES['state_country_map']:
+        candidates = _STATE_RESOURCES['state_country_map'][cleaned_state]
+        candidates = [(candidate, 1.0) for candidate in candidates] if return_scores else candidates
+
+    # Check if we can find a close match (using default threshold of 0.8 (magic number))
+    else:
+        state_match = find_closest_string(cleaned_state, _STATE_RESOURCES['cleaned_trigrams'])
+        if state_match:
+            best_matches, score = state_match
+            candidates = (candidate for state in best_matches
+                          for candidate in _STATE_RESOURCES['state_country_map'][state])
+            candidates = [(candidate, score) for candidate in candidates] if return_scores else list(candidates)
+        else:
+            # No match found
+            candidates = []
+
+    # Return canonical form
+    if return_scores:
+        return sorted((state, _COUNTRY_RESOURCES['cleaned_to_capitilized'][country], score)
+                      for (state, country), score in candidates)
+    else:
+        return sorted((state, _COUNTRY_RESOURCES['cleaned_to_capitilized'][country]) for state, country in candidates)
+
+
+def normalize_city(
+    city: str, return_scores: bool = False
+) -> List[Union[Tuple[List[str], str, float], Tuple[List[str], str]]]:
+    """
+    Cleans up a city by string cleaning and performs some basic city lookups to get the canonical name.
+
+    :param city: City name.
+    :param return_scores: Whether to return match scores.
+    :return: List of (city, country) candidates in canonical form.
+    """
+    # Clean city
+    cleaned_city = clean_city(city)
+
+    # Check if city is part of the known cities list
+    candidates = []
+    not_found = True
+    for country, cities_in_country in _CITY_RESOURCES["cities_per_country_map"].items():
+        # This condition prevent from searching again in country codes
+        if country.upper() not in list(_CITY_RESOURCES["country_code_to_country"].keys()):
```

Comment:
No need to use `list` cast or `.keys()` here. The default way that Python checks if an element is part of a dict object is by looking at the keys. So a simple `if country.upper() not in _CITY_RESOURCES["country_code_to_country"]` is better (and quicker)